File: Bookstore.Core/Models/Orders/OrderItemModel.cs
using Bookstore.Core.Models.PrintingEdition;
using System;
using static Bookstore.Shared.Enums.Enums;

namespace Bookstore.Core.Models.Orders
{
    public class OrderItemModel : Model
    { 
        public Guid PrintingEditionId { get; set; }

        public PrintingEditionModel PrintingEdition { get; set; }

        public Guid OrderId { get; set; }

        public OrderModel Order { get; set; }

        public int Count { get; set; }

        public CurrencyType Currency { get; set; }
    }
}

File: Bookstore.Web/ClientApp/src/app/models/order/order-item-model.ts
import { CurrencyType } from "src/app/enums/currency-type";
import { PrintingEdition } from "../printing-edition/printing-edition.model";
import { Order } from "./order-model";

export class OrderItem
{
  id: string
  count: number
  currency: CurrencyType
  printingEditionId: string
  printingEdition: PrintingEdition
  orderId?: string
  order: Order
}